src/libraries/System.Text.Json/src/System/Text/Json/Nodes/JsonNode.cs

```diff
@@ -311,17 +312,18 @@ public static bool DeepEquals(JsonNode? node1, JsonNode? node2)
         [RequiresDynamicCode(JsonValue.CreateDynamicCodeMessage)]
         public void ReplaceWith<T>(T value)
         {
+            JsonNode? node;
             switch (_parent)
             {
                 case null:
                     return;
```

Comment:
do you need to special case this?

Reply:
Yes, it needs to call different methods depending on whether the parent type is an object or an array.

Reply:
Sorry, I was not clear. I meant do you need `case null` in the switch?

Reply:
Ah yes, that seems redundant.